### ISSUE TITLE
Example REST client

### ISSUE THREAD

**Issue by zloidemon** [tris-foundation/examples#8]
Exaples have to have example of REST client implementation like github module but more detailed with easy to use api.

**Comment by tonyfreeman** [comment]
Just a note: both github and http clients are in early stage, I'm going to add the examples a bit later.

**Comment by tonyfreeman** [comment]
Generally you don't want to use Fiber/Async alongside with regular event loop (like the one Perfect build on) / preemptive multitasking, although it's possible but requires extra work/care/synchronization, this is why I droped synonymous version of `Network` (which is tarantool's iproto depends on, part of AIO now) and switched to fully asynchronous code with no ifdef/else and other "workarounds". 

So there are 2 options:

1) Write your own Async implementation like [this one (deprecated)](https://github.com/tonyfreeman/async-dispatch), where async.wait would block using something like poll / select. (slow/unsafe)
2) Use our [http](https://github.com/tris-foundation/http) (for simple http handler) or more powerfull [web](https://github.com/tris-foundation/web) package both built on top of AIO.

Also you could open an issue in Tarantool repository to implement "independent" connector compatible with the outside world (other swift projects available on github, they all built using Dispatch or the new swift-nio project)

**Comment by tonyfreeman** [comment]
@yurytom I'm going to port your example to our http module, hope to push it in a few minutes.

**Comment by yurytom** [comment]
Thanks again @tonyfreeman 

I will check out your http and web framework. For my pet project I need a web server as a REST API one. That Perfect Swift is probably too much anyway for that task.

Does your http server support SSL? If not, any plans to have it onboard? Any workarounds.

**Comment by tonyfreeman** [comment]
@yurytom yeah, that's the main issue, I was working on TLS until I had to come back to life. I'm going to continue my work (all of this) very soon. For now you can use nginx as a proxy. Here is an example https://github.com/tris-foundation/examples/blob/master/nginx-spa/tris-nginx.conf#L8

**Comment by tonyfreeman** [comment]
@yurytom here is it https://github.com/tris-foundation/examples/tree/master/tarantool/connector-http-server without any tarantool logic yet. Feel free to push some or I'll enhance it later.

**Comment by yurytom** [comment]
@tonyfreeman great, thank you!

**Comment by yurytom** [comment]
Hi @tonyfreeman 

There is a timeout issue when trying to connect to Tarantool. This is based on Tarantool packages from 4.1 branch. This CURL request just hangs.

curl -X POST -H "Content-Type: application/json" "test@email.com", "password": "dfsdf3234@"}' http://localhost:8181/api/v1/registration.

It all works fine if I comment out tarantool connection lines from the code.
Couldn't find anything suspicious in tarantool log even with log level 7

 }
 
```
 func registrationHandler(user: Credentials) -> RegistrationResponse {
    do {
        let iproto = try IProto(host: "127.0.0.1", port: 3301)
        try iproto.auth(username: "admin", password: "admin")
    } catch {
          print("registrationHandler() error: \(error)")
    } 

     return RegistrationResponse(userId: "1")
 }

```

Thanks,
Yury
